src/basekernels/nn.jl

```diff
@@ -37,6 +37,26 @@ function (κ::NeuralNetworkKernel)(x, y)
     return asin(dot(x, y) / sqrt((1 + sum(abs2, x)) * (1 + sum(abs2, y))))
 end
 
+function kernelmatrix(
+    k::NeuralNetworkKernel, x::AbstractVector{<:Real}, y::AbstractVector{<:Real}
+)
+    return kernelmatrix(k, ColVecs(reshape(x, 1, :)), ColVecs(reshape(y, 1, :)))
```

Comment:
Maybe one could add an (internal?) function for `ColVecs(reshape(...))`? A while ago, sometimes I also used `ColVecs(permutedims(x))` (IIRC mainly to replace the (recursive) `x'` which can be problematic sometimes, it seemed).

Reply:
Could you provide an example of the kind of thing that you have in mind? Are you imagining a constructor of the form `ColVecs(::AbstractVector)` or something?

Reply:
No, not sure if that would be a good idea. Maybe it is. But I thought rather some simple internal function for such cases to 1) save typing, 2) consistently use the same approach, and 3) be able to change the approach easily (e.g. if we figure out `permutedims` works better for AD etc.).

Reply:
Ahh okay, so just something like
```julia
_to_row_vec(x::AbstractVector{<:Real}) = reshape(x, 1, :)
```
and
```julia
kernelmatrix(k, ColVecs(_to_row_vec(x)), ColVecs(_to_row_vec(y)))
```
?

Reply:
Or even simpler just
```julia
_to_colvecs(x::AbstractVector{<:Real}) = ColVecs(reshape(x, 1, :))
```
and
```suggestion
kernelmatrix(k, _to_colvecs(x), _to_colvecs(y))
```